pkg/odo/cli/component/create_methods.go

```diff
@@ -60,9 +61,15 @@ func (icm InteractiveCreateMethod) FetchDevfileAndCreateComponent(co *CreateOpti
 		if err != nil {
 			return err
 		}
-	} else {
-		// if the user is logged in or if we have cluster information, display the default project
+	} else if co.KClient != nil {
+		// Get the current project if no --project is passed, but --now is passed
 		componentNamespace = ui.EnterDevfileComponentProject(co.KClient.GetCurrentProjectName())
+	} else {
+		client, _ := kclient.New()
```

Comment:
The error is not checked. Isn't there still a chance that odo will end up with `panic` if if `kclient.New()` returns error and `client` is `nil`?

Reply:
Looking at the `kclient.New()` it does not seem like we would land in this situation since it handles all the errors, but to be on the safe side, I'll switch the change to check if `err == nil`. I hope that works. I do not want to return an error in the case if we're using offline context.